sherpa/models/model.py

```diff
@@ -81,6 +83,48 @@ def cache_model(cls, pars, xlo, *args, **kwargs):
 
 
 class Model(NoNewAttributesAfterInit):
+    """The base class for Sherpa models.
+
+    A model contains zero or more parameters that control the
+    predictions of the model when given one or more coordinates.
+    These parameters may represent some variable that describes
+    the model, such as the temperature of a black body, or the
+    computation, such as what form of interpolation to use.
+
+    Parameters
+    ----------
+    name : str
+        A label for the model instance.
+    pars : sequence of sherpa.parameter.Parameter objects
+        The parameters of the model.
+
+    Attributes
+    ----------
+    name : str
+        The name given to the instance.
+    pars : tuple of sherpa.parameter.Parameter objects
+        The parameters of the model instance.
+
+    Notes
+    -----
+    Parameters can be accessed via the ``pars`` attribute, but it is
+    expected that they will generally be accessed directly, as the
+    class provides case-insensitive access to the parameter names
+    as object attributes. That is, if the model contains parameters
+    called ``breakFreq`` and ``norm``, and the instance is stored in
+    the variable ``mdl``, then the following can be used to access
+    the parameters::
+
+        print("Break frequency = {}".format(mdl.breakfreq))
+
+        mdl.norm = 1.2e-3
+
+    Support is provided to allow parameter names to be changed,
```

Comment:
@olaurino - I think this paragraph (about `_renamedpars`) is from my version of the code. Should it be removed or replaced now we have your implementation?

Reply:
Argh... I thought I had looked for instances of the string `_renamedpars`. Well, at this point the model class could tell the user something about the aliases, but it should probably link to the documentation for `Parameter` to describe the `aliases` argument. The `Parameter` class seems to still have the "old-style" unformatted docstrings, though.

Reply:
My approach has been to try and add relevant documentation when it impacts the code I've been touching, and when I notice the need. I obviously failed to notice the lack of Parameter documentation when writing this ;-)

Reply:
I've removed this paragraph. I'm not sure how best to describe support for parameter aliases, so have left it out for now.